### ISSUE TITLE
Clean-BcContainerDatabase doesn't switch collation.

### ISSUE THREAD

**Issue by ekeukens** [microsoft/navcontainerhelper#1033]
When creating new container with [useNewDatabase] switch the collation should be possible to change (from default). 
When I set a different collation in [Collation.txt] in a local file (in @MyScripts), this file is being copied into the [c:\run\my] path, but the [Clean-BcContainerDatabase] only seems to use the file in [c:\run] path.

```
New-BCContainer `
    -accept_eula `
    -containerName $containername `
    -Auth NavUserPassword `
    -Credential $credential `
    -imageName $imagename `
    -licensefile $LocalLicensePath `
    -updateHosts `
    -myScripts @("C:\temp\Collation.txt") `
    -memoryLimit 20G `
    -multitenant `
    -includeAL `
    -UsecleanDatabase `
    -useNewDatabase `
    -copyTables @('Add-in', 'Report Layout') `
    -additionalParameters @("--volume ${LocalHostSharePath}:${ContainerHostSharePath}", `
                                             "--volume ${LocalAddInsPath}:${ContainerAddInsPath}")

```
**Full output of scripts**
```
NavContainerHelper is version 0.7.0.4
NavContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 1903
Docker Client Version is 19.03.8
Docker Server Version is 19.03.8
Removing container bc154
Removing bc154 from host hosts file
Removing C:\ProgramData\NavContainerHelper\Extensions\bc154
Using image mcr.microsoft.com/businesscentral/sandbox:15.4.41023.43207-nl-ltsc2016
Creating Container bc154
Version: 15.4.41023.43207-nl
Style: sandbox
Platform: 15.0.40939.43184
Generic Tag: 0.0.9.99
Container OS Version: 10.0.14393.3443 (ltsc2016)
Host OS Version: 10.0.18362.900 (1903)
The container operating system does not match the host operating system, forcing hyperv isolation.
Using locale nl-NL
Using hyperv isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\bacpac15\4805272.flf
Additional Parameters:
--volume C:\bacpac15\dockershares\bc154:C:\hostshare\bc154
--volume C:\bacpac15\addins:C:\Run\Add-Ins
Files in C:\ProgramData\NavContainerHelper\Extensions\bc154\my:
- AdditionalOutput.ps1
- Collation.txt
- license.flf
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bc154 from image mcr.microsoft.com/businesscentral/sandbox:15.4.41023.43207-nl-ltsc2016
6a62156e6a0b93a3ce02453b397a6d1ef5401beb165896932af123c7a70eb20d
Waiting for container bc154 to be ready
Initializing...
Setting host.docker.internal to 192.168.192.45 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.192.45 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.27.192.1 in container hosts file
Starting Container
Hostname is bc154
PublicDnsName is bc154
WARNING: Container starts with TimeZone = West-Europa (standaardtijd), which is not recognized in the list of TimeZones.
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 22D214AF14F759E2BD9CCED9F15C00936D3134BB
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.27.202.204
Container Hostname  : bc154
Container Dns Name  : bc154
Web Client          : http://bc154/BC/
Dev. Server         : http://bc154
Dev. ServerInstance : BC
Setting bc154 to 172.27.202.204 in host hosts file

Files:
http://bc154:8080/al-4.0.235643.vsix

WARNING: You are running a container which is 131 days old.
Microsoft recommends that you always run the latest version of our containers.

Initialization took 80 seconds
Ready for connections!
Reading CustomSettings.config from bc154
Creating Desktop Shortcuts for bc154
Creating .net Assembly Reference Folder for VS Code
Copying DLLs from C:\Windows\assembly to assemblyProbingPath
Copying DLLs from C:\Program Files\Microsoft Dynamics NAV\150\Service to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Microsoft Dynamics NAV\150\RoleTailored Client to assemblyProbingPath
Copying DLLs from C:\Test Assemblies\Mock Assemblies to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Open XML SDK to assemblyProbingPath
Stopping ServiceTier in order to replace database
Copying Database on localhost\SQLEXPRESS from Financialsnl to mytempdb
Taking database Financialsnl offline
Copying database files
Attaching files as new Database mytempdb
Putting database Financialsnl back online
Removing Database Financialsnl from localhost\SQLEXPRESS
Creating new database Financialsnl on localhost\SQLEXPRESS with Collation Latin1_General_100_CS_AS
Copying table [Add-in] from original database
Copying table [Report Layout] from original database
Copying table [Entitlement] from original database
Copying table [Entitlement Set] from original database
Copying table [Membership Entitlement] from original database
Removing Database mytempdb from localhost\SQLEXPRESS
Starting Service Tier
Synchronizing
Importing license file
Importing License C:\ProgramData\NavContainerHelper\Extensions\bc154\my\license.flf
Creating user admin
Creating User admin
Assigning Permission Set SUPER to admin
Publishing System Symbols
Copy from container bc154 (C:\Program Files\Microsoft Dynamics NAV\150\AL Development Environment\System.app) to C:\ProgramData\NavContainerHelper\Extensions\bc154\_System.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\bc154\_System.app
App successfully published
Creating Company
Creating company My Company in default
Company successfully created
Publishing System Application
Copy from container bc154 (C:\Applications\System Application\Source\Microsoft_System Application.app) to C:\ProgramData\NavContainerHelper\Extensions\bc154\_Microsoft_System Application.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\bc154\_Microsoft_System Application.app
Synchronizing System Application on tenant default
Installing System Application on tenant default
App successfully published
Switching to multitenant
Copying Database on localhost\SQLEXPRESS from Financialsnl to tenant
Taking database Financialsnl offline
Copying database files
Attaching files as new Database tenant
Putting database Financialsnl back online
Removing Database Financialsnl from localhost\SQLEXPRESS
Exporting Application to Financialsnl
Removing Application from tenant
WARNING: The new settings value will not take effect until you stop and restart the service.
Creating Tenant default on bc154
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting Database for default on server localhost\SQLEXPRESS
Sync'ing Tenant
Tenant successfully created
Container bc154 successfully created

**Screenshots**
If applicable, add screenshots to help explain your problem.

See if I open the container and check the content of the file:

[BC154]: PS C:\Run> cd my

[BC154]: PS C:\Run\my> Get-Content Collation.txt
Latin1_General_100_CI_AS


I think Clean-BcContainerDatabase needs to check which file to use. It seems to use [c:\run\Collation.txt] only.

            $CollationParam = @{}
            if (Test-Path "c:\run\Collation.txt") {
                $Collation = Get-Content "c:\run\Collation.txt"
                $CollationParam = @{ "Collation" = $collation }
                Write-Host "Creating new database $databaseName on $databaseServerInstance with Collation $Collation"
            }
            else {
                Write-Host "Creating new database $databaseName on $databaseServerInstance with default Collation"
            }

**Comment by freddydk** [comment]
I can definitely fix that.
Just to be sure, If you modify your script to:

`-myScripts @("C:\temp\Collation.txt", @{"additionalsetup.ps1" = "move-item -path 'c:\run\my\collation.txt' -destination 'c:\run\collation.txt' -force" } ) 
`
Then everything works?

**Comment by ekeukens** [comment]
Yes it does..
...
Putting database Financialsnl back online
Removing Database Financialsnl from localhost\SQLEXPRESS
Creating new database Financialsnl on localhost\SQLEXPRESS with Collation Latin1_General_100_CI_AS
Copying table [Add-in] from original database
Copying table [Report Layout] from original database
...

So that was the way to toggle with some modifications.. Now I know again! That's what happens if your scripts are so nice Freddy (we forget how to customize with additionalsetup :-) ), thanks!

But I assume you will change standard as well?

**Comment by freddydk** [comment]
Yes, I will fix this in the next containerhelper to look in the my folder.
Thanks for reporting this.

**Comment by freddydk** [comment]
Shipped in 0.7.0.6
